logic/strategy/card_utils_test.go

```diff
@@ -0,0 +1,102 @@
+package strategy
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+)
+
+func isIn(c model.Card, h []model.Card) bool {
+	for _, thisCard := range h {
+		if c == thisCard {
+			return true
+		}
+	}
+	return false
+}
+
+func validateHand(origHand, thisHand []model.Card) bool {
+	for _, c := range thisHand {
+		if !isIn(c, origHand) {
```

Comment:
I notice `isIn` is only used here, but we may be able to remove it and use the same map that we create from `thisHand` in the (currently) next loop.

that is, we could `validateHand` by:
```
// build a map of the cards in `thisHand`, checking that we don't add the same card twice
// for each card in origHand, check if it is in the map we just created
// validate that the number of cards we found in origHand matches the len of thisHand
```

Does that make sense?

Reply:
`thisHand` is a subset of `origHand`, so not every card in `origHand` will be in the `thisHand` map. Do we need to use two maps?

Reply:
nvm I see what you mean now - done!